brew commands fail with 404 on internal API

### `brew doctor` output

```shell
JSON API packages.x86_64_macos.jws.json
Error: curl: (22) The requested URL returned error: 404
Failure while executing; `/usr/bin/env /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/shims/shared/curl --disable --cookie /dev/null --globoff --user-agent Linuxbrew/5.1.11-45-g366923d\ \(Linux\;\ x86_64\ Ubuntu\ 22.04.5\ LTS\)\ curl/7.81.0 --header Accept-Language:\ en --fail --progress-bar --silent --remote-time --output /home/linuxbrew/.cache/Homebrew/api/internal/packages.x86_64_macos.jws.json --location --disable --cookie /dev/null --globoff --show-error --user-agent Linuxbrew/5.1.11-45-g366923d\ \(Linux\;\ x86_64\ Ubuntu\ 22.04.5\ LTS\)\ curl/7.81.0 --header Accept-Language:\ en --fail --progress-bar --silent --compressed --speed-limit 100 --speed-time 5 https://formulae.brew.sh/api/internal/packages.x86_64_macos.jws.json` exited with 22. Here's the output:
curl: (22) The requested URL returned error: 404


Error: Failure while executing; `/usr/bin/env /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/shims/shared/curl --disable --cookie /dev/null --globoff --user-agent Linuxbrew/5.1.11-45-g366923d\ \(Linux\;\ x86_64\ Ubuntu\ 22.04.5\ LTS\)\ curl/7.81.0 --header Accept-Language:\ en --fail --progress-bar --silent --remote-time --output /home/linuxbrew/.cache/Homebrew/api/internal/packages.x86_64_macos.jws.json --location --disable --cookie /dev/null --globoff --show-error --user-agent Linuxbrew/5.1.11-45-g366923d\ \(Linux\;\ x86_64\ Ubuntu\ 22.04.5\ LTS\)\ curl/7.81.0 --header Accept-Language:\ en --fail --progress-bar --silent --compressed --speed-limit 100 --speed-time 5 https://formulae.brew.sh/api/internal/packages.x86_64_macos.jws.json` exited with 22. Here's the output:
curl: (22) The requested URL returned error: 404


Error: Process completed with exit code 1.
```

### Verification

- [x] I ran `brew update` twice and am still able to reproduce my issue.
- [x] My "`brew doctor` output" above says `Your system is ready to brew` or a definitely unrelated `Tier` message.
- [x] This issue's title and/or description do not reference a single formula e.g. `brew install wget`. If they do, open an issue at https://github.com/Homebrew/homebrew-core/issues/new/choose instead.

### `brew config` output

```shell
HOMEBREW_VERSION: 5.1.11-45-g366923d
ORIGIN: https://github.com/Homebrew/brew
HEAD: 366923dbd90cd72f5cde4cf088bbcc7216866a67
Last commit: 32 minutes ago
Branch: main
Core tap HEAD: c658d49ddb8c149e290f40300d0f75a71f4c74d3
Core tap last commit: 7 minutes ago
Core tap JSON: 12 May 11:24 UTC
HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_COLOR: set
HOMEBREW_DEVELOPER: set
HOMEBREW_DOWNLOAD_CONCURRENCY: 8
HOMEBREW_GITHUB_API_TOKEN: set
HOMEBREW_MAKE_JOBS: 4
HOMEBREW_NO_AUTO_UPDATE: set
HOMEBREW_SIMULATE_MACOS_ON_LINUX: set
HOMEBREW_SORBET_RUNTIME: set
HOMEBREW_UPGRADE_AUTO_UPDATES_CASKS: set
HOMEBREW_USE_INTERNAL_API: set
Homebrew Ruby: 4.0.3 => /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/4.0.3/bin/ruby
CPU: quad-core 64-bit zen4
Clang: N/A
Git: 2.54.0 => /bin/git
Curl: 7.81.0 => /bin/curl
Kernel: Linux 6.8.0-1044-azure x86_64 GNU/Linux
OS: Ubuntu 22.04.5 LTS
Host glibc: 2.35
Host libstdc++: 6.0.30
/usr/bin/gcc-12: 12.3.0
/usr/bin/ruby: N/A
Error: Failure while executing; `/usr/bin/env /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/shims/shared/curl --disable --cookie /dev/null --globoff --user-agent Linuxbrew/5.1.11-45-g366923d\ \(Linux\;\ x86_64\ Ubuntu\ 22.04.5\ LTS\)\ curl/7.81.0 --header Accept-Language:\ en --fail --progress-bar --silent --remote-time --output /home/linuxbrew/.cache/Homebrew/api/internal/packages.x86_64_macos.jws.json --location --disable --cookie /dev/null --globoff --show-error --user-agent Linuxbrew/5.1.11-45-g366923d\ \(Linux\;\ x86_64\ Ubuntu\ 22.04.5\ LTS\)\ curl/7.81.0 --header Accept-Language:\ en --fail --progress-bar --silent --compressed --speed-limit 100 --speed-time 5 https://formulae.brew.sh/api/internal/packages.x86_64_macos.jws.json` exited with 22. Here's the output:
curl: (22) The requested URL returned error: 404
```

### What were you trying to do (and why)?

Run `brew test-bot --only-tap-syntax` (style / readall / audit) on a Linux GitHub Actions runner against my [homebrew-php](https://github.com/shivammathur/homebrew-php) third-party tap, using `HOMEBREW_SIMULATE_MACOS_ON_LINUX=1`.


### What happened (include all command output)?

After #22226 and #22033, brew resolves the platform tag and curls `https://formulae.brew.sh/api/internal/packages.<tag>.jws.json`. With `HOMEBREW_SIMULATE_MACOS_ON_LINUX=1` the tag is the synthetic `x86_64_macos` (or `arm64_macos` on arm Linux), and that file is not published:

```
$ curl -sI https://formulae.brew.sh/api/internal/packages.x86_64_macos.jws.json | head -1
HTTP/2 404
$ curl -sI https://formulae.brew.sh/api/internal/packages.arm64_macos.jws.json | head -1
HTTP/2 404
```
As a result

`brew test-bot --only-tap-syntax` exits before any style/readall/audit check runs:

```
✘ JSON API packages.x86_64_macos.jws.json
Error: curl: (22) The requested URL returned error: 404
```

Failing CI run: https://github.com/shivammathur/homebrew-php/actions/runs/25693100272/job/75434357332

### What did you expect to happen?

No errors while running brew commands

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
docker run --rm \
  -e HOMEBREW_DEVELOPER=1 \
  -e HOMEBREW_GITHUB_ACTIONS=1 \
  -e HOMEBREW_NO_AUTO_UPDATE=1 \
  -e HOMEBREW_SIMULATE_MACOS_ON_LINUX=1 \
  ghcr.io/homebrew/ubuntu22.04:main \
  brew test-bot --only-tap-syntax
```

--- a/Library/Homebrew/extend/os/linux/simulate_system.rb
+++ b/Library/Homebrew/extend/os/linux/simulate_system.rb
@@ -8,7 +8,9 @@ module OS
         sig { returns(T.nilable(Symbol)) }
         def os
           @os ||= T.let(nil, T.nilable(Symbol))
-          return :macos if @os.blank? && Homebrew::EnvConfig.simulate_macos_on_linux?
+          if @os.blank? && Homebrew::EnvConfig.simulate_macos_on_linux?
+            return MacOSVersion.new(HOMEBREW_MACOS_NEWEST_SUPPORTED).to_sym
+          end
 
           @os
         end

--- a/Library/Homebrew/test/simulate_system_spec.rb
+++ b/Library/Homebrew/test/simulate_system_spec.rb
@@ -133,10 +133,10 @@ RSpec.describe Homebrew::SimulateSystem do
       expect(described_class.current_os).to eq MacOS.version.to_sym
     end
 
-    it "returns `:macos` on Linux with HOMEBREW_SIMULATE_MACOS_ON_LINUX", :needs_linux do
+    it "returns the newest supported macOS symbol on Linux with HOMEBREW_SIMULATE_MACOS_ON_LINUX", :needs_linux do
       described_class.clear
       ENV["HOMEBREW_SIMULATE_MACOS_ON_LINUX"] = "1"
-      expect(described_class.current_os).to eq :macos
+      expect(described_class.current_os).to eq MacOSVersion.new(HOMEBREW_MACOS_NEWEST_SUPPORTED).to_sym
     end
   end
 end

